eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/serialization/core/Box.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.serialization.core;
+
+import android.os.Parcelable;
+
+
+/**
+ * @author Gabor Keszthelyi
+ */
+public interface Box<T> extends Parcelable
+{
+
+    T content();
```

Comment:
(Note: I start to comment here, but will probably make updates for the other comments)

One thing that have come up now is that `Box<T>` can be used inside `Parcelable` implementations as well, and we don't have `Context` there in the `CREATOR`. 
Example:
```
    public void writeToParcel(Parcel dest, int flags)
    {
        dest.writeParcelable(mDelegate, flags);
        dest.writeParcelable(new DateTimeBox(mDate), flags);
    }

public Dated createFromParcel(Parcel in)
        {
            ClassLoader classLoader = getClass().getClassLoader();
            return new Dated((
                    (DateTimeBox) in.readParcelable(classLoader)).content(),
                    (EventFilter) in.readParcelable(classLoader));
        }
```
(This is in current `serialization.temp.parcelable.Dated` - will write about those later..).